module/VuFind/src/VuFind/RecordDriver/Feature/VersionAwareTrait.php

```diff
@@ -0,0 +1,102 @@
+<?php
+/**
+ * Logic for record versions support.
+ *
+ * PHP version 7
+ *
+ * Copyright (C) The National Library of Finland 2020.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  RecordDrivers
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Site
+ */
+namespace VuFind\RecordDriver\Feature;
+
+/**
+ * Logic for record versions support.
+ *
+ * @category VuFind
+ * @package  RecordDrivers
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Site
+ */
+trait VersionAwareTrait
+{
+    /**
+     * Return count of other versions available
+     *
+     * @return int
+     */
+    public function getOtherVersionCount()
+    {
+        if (null === $this->searchService) {
+            return false;
+        }
+
+        if (!($workKeys = $this->getWorkKeys())) {
```

Comment:
Maybe we should use tryMethod() here instead of getWorkKeys() in case VersionAwareInterface is not implemented. Then before returning false, we could throw an exception if the required interface is not implemented. Or, alternatively, we could just document the dependency in a comment if we want to be less heavy-handed.